### DISCUSSION TITLE
Mixed architecture deployment of Runner scale sets

### DISCUSSION THREAD

**Discussion by danielloader** [actions/actions-runner-controller#3413] [Questions]
Hey all, I'd like to ask what the best practices are for deploying into mixed architecture clusters?

I'm running EKS with Karpenter so I can ask for a node with the [correct pod labels](https://karpenter.sh/docs/concepts/scheduling/#labels). e.g. `kubernetes.io/arch=amd64` or `kubernetes.io/arch=arm64`.

Since I see runner scale sets don't support multiple labels now I can't see a way to easily show one set works with amd64 and another works with arm64.

I'd like to utilise both node types on demand for matrix builds of container images to speed up multi-arch image builds.

An example would look like this: https://docs.docker.com/build/ci/github-actions/multi-platform/#distribute-build-across-multiple-runners.

- Spawn a matrix of jobs:
   - arm64
   - amd64
- Assign them to runners who can take them, and launch pods with the correct `kubernetes.io/arch` pod label.

I'm imagining the solution is just going to be spawning two runner scale sets and add to the pod template, but just wanted to check.

**Comment by wez** [comment]
I have the same question in mind; what did you end up doing here @danielloader?

**Reply by danielloader** [reply]
Simply made two runner sets, amd64 and arm64, not ideal but worked for me.

Ultimately I dropped arm64 runners and just deployed buildkitd on arm 64 and amd64 nodes and just build containers this way and accept the rest of my CI jobs run on amd64 nodes only.
